lib/src/local_working_copy.rs

```diff
@@ -1305,13 +1342,13 @@ impl FileSnapshotter<'_> {
         path: RepoPathBuf,
         disk_path: &Path,
         maybe_current_file_state: Option<&FileState>,
-        new_file_state: FileState,
+        mut new_file_state: FileState,
     ) -> Result<(), SnapshotError> {
         let update = self.get_updated_tree_value(
             &path,
             disk_path,
             maybe_current_file_state,
-            &new_file_state,
+            &mut new_file_state,
         )?;
         if let Some(tree_value) = update {
```

Comment:
I think `new_file_state` can be updated here instead of passing mutable ref around. Something like `if normal_file && (None | !is_resolved()) { new_file_state.materialized_conflict_data = .. }`.

The downstream functions shouldn't be responsible for the new `FileState`.

(We might have to rewrite some `current_file_state == new_file_state` to not compare uninteresting fields)

Reply:
That's a good idea, thanks! I updated it based on your comments. Could you take another look and let me know if there's anything I should change?

Reply:
LG, thanks!